### PR TITLE
Switch the `pkgs/by-name` check to a separate repository

### PR DESCRIPTION
**Pull request by infinisil**
## Description of changes

This completes the move of the `pkgs/by-name` tooling to the new https://github.com/NixOS/nixpkgs-check-by-name repository, see https://github.com/NixOS/nixpkgs/issues/286559 for motivation.

> [!Note]
> I'm leaving in both `pkgs/test/nixpkgs-check-by-name` to make it easier to review the removal in a separate PR, and `pkgs/test/nixpkgs-check-by-name/scripts/pinned-tool.json` because it's needed for CI of this PR to succeed.

This work is sponsored by [Antithesis](https://antithesis.com/) :sparkles:

## Things done
- [x] Test it in CI: https://github.com/tweag/nixpkgs/pull/87
- [x] Test it locally: `./maintainers/scripts/check-by-name.sh master`
- [x] Update all comments (especially in the workflow)

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/README.md [hunk 1]
```diff
@@ -108,13 +108,13 @@ There's some limitations as to which packages can be defined using this structur
 
 ## Validation
 
-CI performs [certain checks](../test/nixpkgs-check-by-name/README.md#validity-checks) on the `pkgs/by-name` structure.
-This is done using the [`nixpkgs-check-by-name` tool](../test/nixpkgs-check-by-name).
+CI performs [certain checks](https://github.com/NixOS/nixpkgs-check-by-name?tab=readme-ov-file#validity-checks) on the `pkgs/by-name` structure.
```
Comment (philiptaron) [inline pkgs/by-name/README.md:111]: Future: we should publish with GitHub pages so that https://nixpkgs-check-by-name.github.io/ is the URL.

**Review by philiptaron — APPROVED** [review]
Looks good.

**Comment by infinisil** [comment]
Drafted because I noticed a minor problem that would occur with the follow-up removal PR, where I wanted to be able to remove `pkgs/test/nixpkgs-check-by-name` but persist its `./scripts` in `pkgs/test/check-by-name`. Would cause a CI failure though unless I'd delay moving `pkgs/test/nixpkgs-check-by-name/scripts/pinned-version.txt` for _another_ PR. I'll do the alternative of moving it now, so that the follow-up PR will _only_ remove things.

**Comment by infinisil** [comment]
Force pushed with the necessary changes [(diff)](https://github.com/NixOS/nixpkgs/compare/177fc1a8e50bf442af477b5153f64e1ac957cc75..f7ea336cb2bd403bb0bc8ce9ce48479a1427de18)

File: pkgs/by-name/README.md (hunk 1 above)
Comment (infinisil) [reply pkgs/by-name/README.md:111]: Doesn't seem necessary, but yeah that's an option :)

**Review by philiptaron — APPROVED** [review]
Yeah, let's go x2

**Pull request merged**